src/sodaspark/scan.py

```diff
@@ -296,4 +303,136 @@ def execute(
     df.createOrReplaceTempView(scan_yml.table_name)
     scan = create_scan(scan_yml, soda_server_client=soda_server_client)
     scan.execute()
+    if as_frame:
+        return convert_scan_result_to_spark_data_frames(scan.scan_result)
     return scan.scan_result
+
+
+def measurements_to_data_frame(measurements: list[Measurement]) -> DataFrame:
+    """
+    Convert measurements to a data frame.
+    Parameters
+    ----------
+    measurements: List[Measurement]
+        The measurements.
+    Returns
+    -------
+    out : DataFrame
+        The measurements as data frame.
+    """
+    schema_group_values = T.StructType(
+        [
+            T.StructField(
+                "group", T.MapType(T.StringType(), T.StringType()), True
+            ),
+            T.StructField("value", T.StringType(), True),
+        ]
+    )
+    schema = T.StructType(
+        [
+            T.StructField("metric", T.StringType(), True),
+            T.StructField("column_name", T.StringType(), True),
+            T.StructField("value", T.StringType(), True),
+            T.StructField(
+                "group_values", T.ArrayType(schema_group_values, True), True
+            ),
+        ]
+    )
+    spark_session = SparkSession.builder.getOrCreate()
+    out = spark_session.createDataFrame(
+        measurements,
+        schema=schema,
+    )
+    return out
+
+
+def testresults_to_data_frame(testresults: list[TestResult]) -> DataFrame:
+    """
+    Convert TestResults to a data frame.
+    Parameters
+    ----------
+    testresults: List[TestResult]
+        The testresults.
+    Returns
+    -------
+    out : DataFrame
+        The testresults as data frame.
+    """
+
+    schema_test = T.StructType(
+        [
+            T.StructField("column", T.StringType(), True),
+            T.StructField("expression", T.StringType(), True),
+            T.StructField("id", T.StringType(), True),
+            T.StructField("metrics", T.ArrayType(T.StringType(), True), True),
+            T.StructField("title", T.StringType(), True),
+        ]
+    )
+
+    schema = T.StructType(
+        [
+            T.StructField("test", schema_test, True),
+            T.StructField("passed", T.BooleanType(), True),
+            T.StructField("skipped", T.BooleanType(), True),
+            T.StructField(
+                "values", T.MapType(T.StringType(), T.StringType()), True
+            ),
+            T.StructField("error", T.StringType(), True),
+            T.StructField(
+                "group_values", T.MapType(T.StringType(), T.StringType()), True
+            ),
+        ]
+    )
+    spark_session = SparkSession.builder.getOrCreate()
+    out = spark_session.createDataFrame(
+        testresults,
+        schema=schema,
+    )
+    return out
+
+
+def scanerror_to_data_frame(scanerrors: list[ScanError]) -> DataFrame:
```

Comment:
also `scanerror` is two words: `scan_errors`. And in the function title make is plural: scanerror**s**

Reply:
Updated to `scan_errors` in both title and argument